src/mesh_analysis.rs

```diff
@@ -0,0 +1,413 @@
+use std::collections::HashSet;
+
+use crate::edge_analysis::EdgeCountMap;
+use crate::geometry::{Geometry, OrientedEdge};
+
+/// Check if all the vertices of geometry are referenced in geometry's faces
+#[allow(dead_code)]
+pub fn has_no_orphans(geo: &Geometry) -> bool {
```

Comment:
This is already implemented on `Geometry`, why have it here as well?

Reply:
this function was moved one of the previous PRs, probably got forgotten here.